pyccl/halos/profiles/IA.py

```diff
@@ -0,0 +1,426 @@
+        if lmax is not None:
+            self.lmax = lmax
+        if log10Mmin_0 is not None:
+            self.log10Mmin_0 = log10Mmin_0
+        if log10Mmin_p is not None:
+            self.log10Mmin_p = log10Mmin_p
+        if log10M0_0 is not None:
+            self.log10M0_0 = log10M0_0
+        if log10M0_p is not None:
+            self.log10M0_p = log10M0_p
+        if log10M1_0 is not None:
+            self.log10M1_0 = log10M1_0
+        if log10M1_p is not None:
+            self.log10M1_p = log10M1_p
+        if siglnM_0 is not None:
+            self.siglnM_0 = siglnM_0
+        if siglnM_p is not None:
+            self.siglnM_p = siglnM_p
+        if alpha_0 is not None:
+            self.alpha_0 = alpha_0
+        if alpha_p is not None:
+            self.alpha_p = alpha_p
+        if bg_0 is not None:
+            self.bg_0 = bg_0
+        if bg_p is not None:
+            self.bg_p = bg_p
+        if bmax_0 is not None:
+            self.bmax_0 = bmax_0
+        if bmax_p is not None:
+            self.bmax_p = bmax_p
+        if a_pivot is not None:
+            self.a_pivot = a_pivot
+        if ns_independent is not None:
+            self.ns_independent = ns_independent
+        if rmin is not None:
+            self.rmin = rmin
+        if N_r is not None:
+            self.N_r = N_r
+        if N_jn is not None:
+            self.N_jn = N_jn
+
+    def _I_integral(self, a, b):
+        '''
+        Computes the integral
+        .. math::
+            I(a,b) = \\int_{-1}^1 \\mathrm{d}x (1-x^2)^{a/2}x^b =
+            \\frac{((-1)^b+1)\\Gamma(a+1)\\Gamma\\left
+            \\frac{b+1}{2}\\right}
+            {2\\Gamma\\left(a+\\frac{b}{2}+\\frac{3}{2}\\right}.
+        '''
+        from scipy.special import gamma
+        return (1+(-1)**b)*gamma(a/2+1)*gamma((b+1)/2)/(2*gamma(a/2+b/2+3/2))
+
+    def _fl(self, l, thk=np.pi / 2, phik=None, b=-2):
+        '''
+        Computes the angular part of the satellite intrinsic shear field,
+        Eq. (C8) in Fortuna et al. 2021.
+        '''
+        from scipy.special import binom
+        gj = np.array([0, 0, np.pi / 2, 0, np.pi / 2, 0, 15 * np.pi / 32,
+                       0, 7 * np.pi / 16, 0, 105 * np.pi / 256, 0,
+                       99 * np.pi / 256])
+        l_sum = 0.
+        if b == 0:
+            var1_add = 1
+        else:
+            var1_add = b
+        for m in range(0, l + 1):
+            m_sum = 0.
+            for j in range(0, m + 1):
+                m_sum += (binom(m, j) * gj[j] *
+                          self._I_integral(j + var1_add, m - j) *
+                          np.sin(thk) ** j * np.cos(thk) ** (m - j))
+            l_sum += binom(l, m) * binom((l + m - 1) / 2, l) * m_sum
+        if phik is not None:
+            l_sum *= np.exp(1j * 2 * phik)
+        return 2 ** l * l_sum
+
+    def get_normalization(self, cosmo, a, hmc):
+        """Returns the normalization of this profile, which is the
+        mean galaxy number density.
```

Comment:
Currently testing this normalization but this seems to not be doing the right thing. Also, in the `_get_prefactor` method in [my PR](https://github.com/LSSTDESC/CCL/blob/3ed354cbfcd2cfbb51f8ce6ccf4f4bd1d480d998/pyccl/halos/profiles.py#L1871) the normalization is 1/n_gal.

Reply:
Here it's n_gal (the galaxy number density).

Reply:
Okay, I have run some more tests and this does what it should do, but now the user has to explicitly request the normalization of the profiles (which is the correct thing to do). I also noticed this will be the default option in v3, correct?